packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+class _CupertinoTableViewDivider extends StatelessWidget {
+  const _CupertinoTableViewDivider({
+    this.child,
+  });
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return CustomPaint(
+      foregroundPainter: _CupertinoTableViewDividerPainter(),
+      child: child,
+    );
+  }
+}
+
+class _CupertinoTableViewDividerPainter extends CustomPainter {
+  _CupertinoTableViewDividerPainter({
+    this.dividerThickness = kDefaultDividerHeight,
+    this.dividerInset = kDefaultCellInset,
+    this.dividerColor = CupertinoColors.lightBackgroundGray,
+  }) : _dividerPaint = Paint()
+    ..color = dividerColor
+    ..style = PaintingStyle.fill;
+
+  final double dividerThickness;
+  final double dividerInset;
+  final Color dividerColor;
+  final Paint _dividerPaint;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    canvas.drawRect(
+      Rect.fromLTRB(
+        dividerInset,
+        size.height - dividerThickness,
+        size.width,
+        size.height,
+      ),
+      _dividerPaint,
+    );
+  }
+
+  @override
+  bool shouldRepaint(_CupertinoTableViewDividerPainter oldPainter) {
+    return  dividerThickness != oldPainter.dividerThickness ||
+        dividerColor != oldPainter.dividerColor ||
+        dividerInset != oldPainter.dividerInset;
+  }
+
+}
+
+/// Widget that renders a [Sliver] that draws extra dividers after the last cell
+/// in a [CupertinoTableView].
+///
+/// No dividers will be drawn if [CupertinoTableView] has enough cells to scroll,
+/// as per iOS behavior.
+///
+/// The gap between extra dividers is equal to the height of the last cell in
+/// the [CupertinoTableView], as per iOS behavior.
+@visibleForTesting
+class CupertinoTableViewExtraDividers extends RenderObjectWidget {
+  const CupertinoTableViewExtraDividers({
+    double cellHeight,
+  }) : cellHeight = cellHeight ?? kDefaultTableViewCellHeight;
+
+  final double cellHeight;
+
+  @override
+  RenderObjectElement createElement() {
+    return CupertinoTableViewExtraDividersElement(this);
+  }
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return CupertinoTableViewExtraDividersRenderObject(
+      cellHeight: cellHeight,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, CupertinoTableViewExtraDividersRenderObject renderObject) {
+    renderObject.cellHeight = cellHeight;
+  }
+}
+
+@visibleForTesting
+class CupertinoTableViewExtraDividersElement extends RenderObjectElement {
+  CupertinoTableViewExtraDividersElement(Widget widget) : super(widget);
+
+  @override
+  void forgetChild(Element child) {}
+
+  @override
+  void insertChildRenderObject(RenderObject child, dynamic slot) {}
+
+  @override
+  void moveChildRenderObject(RenderObject child, dynamic slot) {}
+
+  @override
+  void removeChildRenderObject(RenderObject child) {}
+
+}
+
+@visibleForTesting
+class CupertinoTableViewExtraDividersRenderObject extends RenderSliver {
```

Comment:
I would just use test/rendering/mock_canvas.dart

Reply:
what does that mean?

Reply:
See https://github.com/flutter/flutter/blob/master/packages/flutter/test/cupertino/scrollbar_test.dart for instance. Rather than making various layers in between public (well the painter is public because it's shared between material and ios), it black box tests how it paints and asserts that's correct.